onnx/defs/tensor/utils.cc

```diff
@@ -98,4 +98,73 @@ void resizeShapeInference(InferenceContext& ctx, bool is_resize_op) {
     }
   } // nullptr != scales
 }
+
+void resizeShapeInferenceHelper_opset7(
+    const TensorShapeProto& input_shape,
+    const std::vector<float>& scales_data,
+    TensorShapeProto* output_shape) {
+  for (int i = 0; i < input_shape.dim_size(); ++i) {
+    auto* dim = output_shape->mutable_dim(i);
+    // If input_shape has dim_value, we caculate the scaled result
+    // If input_shape doesn's have one, we leave it here
+    if (input_shape.dim(i).has_dim_value()) {
+      int64_t dim_value = static_cast<int64_t>(std::floor(
+          static_cast<float>(input_shape.dim(i).dim_value()) * scales_data[i]));
+      // If output_shape has dim_value, we validate the caculated result
+      // If output_shape doesn's have one, we set it to the scaled result
+      if (dim->has_dim_value()) {
+        if (static_cast<int64_t>(dim->dim_value()) != dim_value) {
+          fail_shape_inference(
+              "Dimension value inferred (",
+              dim_value,
+              ") is not equal to the existing dim value (",
+              dim->dim_value(),
+              ").");
+        }
+      } else {
+        dim->set_dim_value(static_cast<int64_t>(dim_value));
+      } // dim->has_dim_value()
+    } // input_shape.dim(i).has_dim_value()
+  }
+}
+
+void resizeShapeInference_opset7(InferenceContext& ctx) {
+  if (!hasNInputShapes(ctx, 1)) {
+    return;
+  }
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
```

Comment:
move this line before the 'if' condition ? We can do TypeInference even if we can't proceed with ShapeInference

Reply:
sure, that makes sense.
Note this PR is just reverting back the shape inference function for these ops < opset 11. I would prefer keeping the Upsample-7 the way it was to reduce the risk of another regression.

Reply:
sure Bowen :)